src/SessionSetup/include/SessionSetup/ConnectToTargetDialog.h

```diff
@@ -64,6 +65,8 @@ class ConnectToTargetDialog : public QDialog {
   std::unique_ptr<orbit_client_services::ProcessManager> process_manager_;
   std::optional<TargetConfiguration> target_configuration_;
 
+  std::unique_ptr<orbit_metrics_uploader::ScopedMetric> connection_metric_;
```

Comment:
Did you consider using an std::optional instead of unique_ptr. It would save a heap allocation